content/en/docs/ops/best-practices/security/index.md

```diff
@@ -129,13 +129,72 @@ For example, if you choose the `DECODE_AND_MERGE_SLASHES` option, you modify the
+### Mitigation for unsupported normalization
 
-#### Case Normalization
+This section describes various mitigation for unsupported normalization. These could be useful when you find a specific
+normalization that is not supported by Istio.
+
+Please make sure you understand the mitigation thoroughly and use it carefully. Some mitigation is just best-effort workaround
+that is not well tested or rely on things that are not supported by Istio.
+
+#### Safer Authorization Policy Pattern
+
+Use the `ALLOW-with-positive-matching` or `DENY-with-negative-matching` patterns whenever possible. These authorization policy
+patterns are safer because the worst result in the case of policy mismatch is just an unexpected 403 rejection instead of
+an authorization policy bypass.
+
+The `ALLOW-with-positive-matching` pattern is to use the `ALLOW` action only with **positive** matching fields (e.g. `paths`, `values`)
+and do not use any of the **negative** matching fields (e.g. `notPaths`, `notValues`).
+
+The `DENY-with-negative-matching` pattern is to use the `DENY` action only with **negative** matching fields (e.g. `notPaths`, `notValues`)
+and do not use any of the **positive** matching fields (e.g. `paths`, `values`).
+
+For example, the authorization policy below uses the `ALLOW-with-positive-matching` pattern to allow requests at path `/public`:
+
+{{< text yaml >}}
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: foo
+spec:
+  action: ALLOW
+  rules:
+  - to:
+    - operation:
+        paths: ["/public"]
+{{< /text >}}
+
+The above policy explicitly lists the allowed path (`/public`). This means the request path must be exactly the same as
+`/public` to let the request allowed by the policy. Any other requests will be rejected by default eliminating the risk
+of unknown normalization behavior causing policy bypass.
+
+The following is another example using the `DENY-with-negative-matching` pattern to achieve the same result:
+
+{{< text yaml >}}
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: foo
+spec:
+  action: DENY
+  rules:
+  - to:
+    - operation:
+        notPaths: ["/public"]
+{{< /text >}}
+
+#### Custom normalization logic
+
+You can apply custom normalization logic using the WASM or Lua filter. It is recommended to use the WAM filter because
+it's officially supported and also used by Istio. The Lua filter could also be used if you need a quick proof-of-concept
+but please be aware the Lua filter is not supported by Istio.
+
+##### Example custom normalization (case normalization)
 
 In some environments, it may be useful to have paths in authorization policies compared in a case insensitive manner.
 For example, treating `https://myurl/get` and `https://myurl/GeT` as equivalent.
-In those cases, the `EnvoyFilter` shown below can be used.
+
+In those cases, the `EnvoyFilter` shown below can be used to insert a Lua filter to normalize the path to lower case.
```

Comment:
If the Lua filter is not officially supported by Istio, how do we claim this solution can be used trouble-free in production? Is this just a best-effort fix?

Reply:
the Lua filter is for quick proof-of-concept, not production traffic.

Reply:
Should be remove the Lua filter and replace it with an Envoy filter so we are showing a supported solution? Or is an Envoy Filter not something we can document here.

Reply:
The Lua filter was added as a mitigation of the previous CVE (Goldpanda), I think we could keep it there until we have the WASM demo ready.